bindings/core/src/method_handler/secret_manager.rs

```diff
@@ -1,63 +1,45 @@
 // Copyright 2023 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-#[cfg(feature = "ledger_nano")]
-use iota_sdk::client::secret::ledger_nano::LedgerSecretManager;
-#[cfg(feature = "stronghold")]
-use iota_sdk::client::secret::{stronghold::StrongholdSecretManager, SecretManager};
+use crypto::signatures::secp256k1_ecdsa::EvmAddress;
 use iota_sdk::{
     client::{
-        api::{GetAddressesOptions, PreparedTransactionData},
-        secret::{DowncastSecretManager, SecretManage, SignBlock},
+        api::PreparedTransactionData,
+        secret::{
+            types::EvmSignature, BlockSignExt, DowncastSecretManager, Generate, SecretManager, Sign, SignTransaction,
+        },
     },
     types::{
-        block::{address::ToBech32Ext, core::UnsignedBlock, unlock::Unlock, BlockDto},
+        block::{
+            address::{Ed25519Address, ToBech32Ext},
+            core::UnsignedBlock,
+            BlockDto,
+        },
         TryFromDto,
     },
 };
 
 use crate::{method::SecretManagerMethod, response::Response, Result};
 
 /// Call a secret manager method.
-pub(crate) async fn call_secret_manager_method_internal<S: SecretManage + DowncastSecretManager>(
-    secret_manager: &S,
+pub(crate) async fn call_secret_manager_method_internal(
+    secret_manager: &SecretManager,
```

Comment:
Why is this not the trait anymore?

Reply:
Since we have to store it, it has to be the enum anyway so I figured we should just use it here. Still, we could change it if it is necessary to support custom SMs

Reply:
We don't need to store it? But yes we need deserialize unless we change something to support custom SMs :+1:

Reply:
Sorry, I meant we need to define the possible variants in the bindings.

Reply:
Worth to keep the trait then for this potential future? If not feel free to keep as it is